app/views/documents/index.html.haml

```diff
@@ -88,9 +87,6 @@
 :javascript
   $("#category").select2();
 
-  /* $("#category").select2({
-    initSelection: function (elm, callback) {
-      callback({id: elm.data('initValue')});
-    },
-    data: null
-  }); */
+  preselected_values = #{params[:q][:with_all_tags] || []};
+
+  $('#category').select2('val', preselected_values);
```

Comment:
this should work but it is not clean solution.
The right way is to put default/selected value to the data attribute. Then use custom selector to enable select2 and set the value.
Using `:javascript` in the html make the code smell. Even it was probably me, who wrote it .-)